kroxylicious-filters/kroxylicious-encryption/src/test/java/io/kroxylicious/filter/encryption/EnvelopeEncryptionFilterTest.java

```diff
@@ -191,42 +184,12 @@ void shouldOnlyEncryptTopicWithKeyId() {
 
         // Then
         verify(keyManager).encrypt(any(), anyInt(), any(),
-                argThat(records -> assertThat(records)
+                argThat(records -> assertThat(records.records())
                         .hasSize(1)
                         .allSatisfy(record -> assertThat(record.value()).isEqualTo(ByteBuffer.wrap(HELLO_CIPHER_WORLD)))),
                 any());
     }
 
-    @Test
-    void produceShouldMaintainClientOffsets() {
```

Comment:
this is now a KeyManager responsibility, I added an assertion to check that non consecutive offsets are preserved.